### PR TITLE
Update azure_rm_aks.py

### PR DESCRIPTION
**Pull request by iRootkit**
##### SUMMARY
Add a vnet_subnet_id parameter to the module docs, a description and note.

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME
agent profile changes docs

```

### CODE REVIEW

**Comment by MyronFanQiu** [comment]
@iRootkit Thanks for the PR. Please fix the CI issue. Based on my understanding, the error may be caused by  `Note:`. It will be treated as a deeper dict in the yml file.

**Comment by MyronFanQiu** [comment]
@zikalino @yungezz Please review this PR when you are free. It's a Docs Pull Request. 
https://github.com/ansible/ansible/blob/6c6f12f893f7aa8393a23ba61e2edd4e1845c663/lib/ansible/modules/cloud/azure/azure_rm_aks.py#L423

File: lib/ansible/modules/cloud/azure/azure_rm_aks.py [hunk 1]
```diff
@@ -77,6 +77,10 @@
             os_disk_size_gb:
                 description:
                     - Size of the OS disk.
+            vnet_subnet_id:
+                description:
+                    - The Subnet ID for a existing Subnet or if you use a different CIDR for the Nodes IP (Kubenet or Azure CNI)
```
Comment (acozine) [inline lib/ansible/modules/cloud/azure/azure_rm_aks.py:82]: I find this confusing. I've made a suggestion as a way to start the conversation - let me know if it's accurate.

```suggestion
                    - The subnet for the agent pool profile. For existing subnet, use subnet ID. For new subnet, use CIDR notation.
```

File: lib/ansible/modules/cloud/azure/azure_rm_aks.py [hunk 2]
```diff
@@ -77,6 +77,10 @@
...
+                    - Note: Does not overlap 169.254.0.0/16, 172.30.0.0/16, 172.31.0.0/16, 192.0.2.0/24
```
Comment (acozine) [inline lib/ansible/modules/cloud/azure/azure_rm_aks.py:83]: I think removing the `:` will fix the CI failure. However, I find this line confusing too. Are the ranges shown here examples of what not to do? What to do? Could we remove them altogether?

```suggestion
                    - Cannot overlap with other address ranges within the same virtual network.
```

**Review by acozine — CHANGES_REQUESTED** [review]
Thanks @iRootkit for adding documentation for a missing sub-option. 

Can you provide a bit more context or information? Will this module create a new subnet if the `vnet_subnet_id` does not exist? If the user specifies a value that does overlap, what happens?

**Comment by Fred-sun** [comment]
@iRootkit Could you changed this PR according by the above comment? when you finished changed, I will push for review! Thank you very much!

**Comment by Fred-sun** [comment]
kindly ping

**Comment by Fred-sun** [comment]
@iRootkit Are you still here? Please update when you're free! Thanks a lot!

**Comment by Fred-sun** [comment]
@iRootkit Please help update this PR when you're free! Thank you very much!

**Comment by Fred-sun** [comment]
kindly ping

**Comment by Fred-sun** [comment]
@iRootkit Are you still here? Can you update PR according to the comments above? Or do you give me permission to update on your basis? Thank you very much!

**Comment by Fred-sun** [comment]
@iRootkit kindly ping！

**Comment by Fred-sun** [comment]
@iRootkit Thank you for taking the time to contribute to this PR. We will transfer ansible 's azure module related Issue and PR to azure collection (https://github.com/ansible-collections/azure/pulls), can you transfer the Issue to azure collection repo?

**Comment by Akasurde** [comment]
Hi @iRootkit, Thank you very much for your interest in Ansible. This plugin/module is no longer maintained in this repository and has been migrated to https://github.com/ansible-collections/azure<br><br>If you have further questions please stop by IRC or the mailing list:<br><br>* IRC: #ansible on irc.freenode.net<br>* mailing list: https://groups.google.com/forum/#!forum/ansible-project<br><br>needs_info<br>

**Pull request closed without merging**